conformance/lang/expressions/literals/int_literal.balt

```diff
@@ -0,0 +1,502 @@
+
+public function main() {
+    any x = 0d;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+
+    x = 1234567890D;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+
+    x = 786.218d;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+}
+
+Test-Case: error
+Description: Test the basic type when the numeric-literal includes a DecimalTypeSuffix.
+Labels: DecimalTypeSuffix, decimal, float, int
+
+public function main() {
+    int x = 0d; // @error expected an 'int' but found a 'decimal'
+
+    float y = 12345678.90D; // @error expected an 'float' but found a 'decimal'
+
+    int z = 786d; // @error expected an 'int' but found a 'decimal'
+}
+
+Test-Case: output
+Description: Test the basic type when the numeric-literal is a HexFloatingPointLiteral.
+Labels: any, decimal, float, HexFloatingPointLiteral, int, is-expr
+
+public function main() {
+    any x = 0xabc12.def;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    x = 0x.2131ef;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    x = 0x123.456cP4;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+}
+
+Test-Case: error
+Description: Test the basic type when the numeric-literal is a HexFloatingPointLiteral.
+Labels: decimal, float, HexFloatingPointLiteral, int
+
+public function main() {
+    decimal x = 0xabc12.def; // @error expected an 'decimal' but found a 'float'
+
+    decimal y = 0x.2131ef; // @error expected an 'decimal' but found a 'float'
+
+    int z = 0x123.456cP4; // @error expected an 'int' but found a 'float'
+}
+
+Test-Case: output
+Description: Test the basic type when the applicable contextually expected type is a subtype of decimal.
+Labels: DecimalNumber, decimal, int, is-expr, nil-type, string
+
+public function main() {
+    decimal a = 1.212;
+    io:println(a is decimal); // @output true
+
+    int|decimal b = 1.212;
+    io:println(b is int); // @output false
+    io:println(b is decimal); // @output true
+
+    int|decimal c = 0.0;
+    io:println(c is int); // @output false
+    io:println(c is decimal); // @output true
+
+    string|decimal d = 32.12;
+    io:println(d is decimal); // @output true
+
+    ()|decimal e = 10;
+    io:println(e is decimal); // @output true
+}
+
+Test-Case: output
+Description: Test the basic type when the applicable contextually expected type is a subtype of float.
+Labels: DecimalNumber, decimal, float, HexIntLiteral, int, nil-type, is-expr, string
+
+public function main() {
+    int|float x = 1.212;
```

Comment:
Wouldn't these tests come under the floating-point-literal tests @suleka96 is writing?

Anyway I think we should move these and whatever we write for floating point literals (related to determining the basic type of a numeric literal) to a separate file named like numeric_literal.bal, because this type-related part is actually a separate part.

Reply:
+1. Will move them into a new `numeric_literal.balt` file.